### PR TITLE
Clean up x/env/OMRCPU class and add missing supportsHLE()

### PR DESCRIPTION
**Pull request by harryyu1994**
- Remove unused parameter (TR::Compilation *comp)
- Add missing supportsHLE() to TR_X86ProcessorInfo
- Remove J9_PROJECT_SPECIFIC macro in OMR::X86::CPU::queryX86TargetCPUID()

Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Is this a WIP because it's not ready for review, or because you want to hold off merging until your downstream change is ready to merge?  If it's the latter, I'll review.

**Comment by harryyu1994** [comment]
Hi Daryl, thanks for looking at this. It's the latter, please review for me.

**Comment by harryyu1994** [comment]
This code change is ready for review and merging.

File: compiler/x/codegen/OMRCodeGenerator.hpp [hunk 1]
```diff
@@ -170,6 +170,7 @@ struct TR_X86ProcessorInfo
    bool supportsSelfSnoop()                {return _featureFlags.testAny(TR_SelfSnoop);}
    bool supportsTM()                       {return _featureFlags8.testAny(TR_RTM);}
    bool supportsHyperThreading()           {return _featureFlags.testAny(TR_HyperThreading);}
+   bool supportsHLE()                      {return _featureFlags8.testAny(TR_HLE);}
```
Comment (mstoodle) [inline compiler/x/codegen/OMRCodeGenerator.hpp]: This change seems unrelated to the commit; please put it in a separate commit.

File: compiler/x/env/OMRCPU.cpp [hunk 2]
```diff
@@ -26,11 +26,14 @@
...
+
 TR_X86CPUIDBuffer *
-OMR::X86::CPU::queryX86TargetCPUID(TR::Compilation *comp)
+OMR::X86::CPU::queryX86TargetCPUID()
    {
 #ifdef J9_PROJECT_SPECIFIC
-   return NULL;
+   static TR_X86CPUIDBuffer buf = { {'U','n','k','n','o','w','n','B','r','a','n','d'} };
+   jitGetCPUID(&buf);
+   return &buf;
```
Comment (mstoodle) [inline compiler/x/env/OMRCPU.cpp]: This change does not seem directly related to removing the comp parameter. Why is this J9 specific code provided here rather than in the OpenJ9 project (https://github.com/eclipse/openj9/blob/master/runtime/compiler/trj9/x/env/J9CPU.cpp) ? It seems like there's an opportunity to clean up this particular function...

**Comment by mstoodle** [comment]
@genie-omr build all

**Comment by mstoodle** [comment]
bah, i only meant to initiate tests on x :( . oh well, it will be over-the-top tested...

**Comment by harryyu1994** [comment]
Hi Mark, @mstoodle thanks for the review, appreciate your help! I would like to hold off merging until I verify that the downstream change(openj9) works as intended. I will let you know.

**Comment by mstoodle** [comment]
fair enough! i marked the dependent PR in Openj9 as depending on this one.

**Comment by harryyu1994** [comment]
Just got back to this. Removing the J9_PROJECT_SPECIFIC guard in OMR::X86::queryX86TargetCPUID() is causing the OpenJ9 build to fail, so I'm planning to leave it for now. I've ran sanity tests with the remaining two commits along with the downstream changes and they seem to be passing. @mstoodle please help me  review and merge when you have time. Downstream change is also ready for review. Thanks in advance!

**Comment by mstoodle** [comment]
@harryyu1994 can you be more specific about how the OpenJ9 build failed?

**Comment by harryyu1994** [comment]
@mstoodle When we execute the JVM, we get `JVMJ9VM011W Unable to load j9jit29: /bluebird/builds/bld_386005/sdk/xa6490/lib/amd64/default/libj9jit29.so: undefined symbol: _ZN2TR9JitConfig8instanceEv`  -----> probably have something to do with `auto jitConfig = TR::JitConfig::instance();` in `OMR::X86::queryX86TargetCPUID()`.

**Comment by mstoodle** [comment]
@harryyu1994 I actually wasn't suggesting to remove the #ifdef, I was proposing to move the #ifdef controlled code into the J9 specific extension of this class (which already exists): i.e. put this #ifdef'd code into J9::X86::CPU::queryX86TargetCPUID() in the openj9 project and remove it from OMR.

Does that make sense?

**Comment by harryyu1994** [comment]
@mstoodle Sorry I didn't quite get it. Here are the two versions of `queryX86TargetCPUID()`, you are saying we should move the code in OMR::queryX86TargetCPUID() to J9::queryX86TargetCPUID(), then what is OMR::queryX86TargetCPUID() left with? Could you please elaborate more on this?

```
OMR::X86::CPU::queryX86TargetCPUID()
    {
 #ifdef J9_PROJECT_SPECIFIC
    return NULL;
 #else
    static TR_X86CPUIDBuffer *buf = NULL;
    auto jitConfig = TR::JitConfig::instance();
 
    if (jitConfig && jitConfig->getProcessorInfo() == NULL)
       {
       buf = (TR_X86CPUIDBuffer *) malloc(sizeof(TR_X86CPUIDBuffer));
       if (!buf)
          return 0;
       jitGetCPUID(buf);
       jitConfig->setProcessorInfo(buf);
       }
    else
       {
       if (!buf)
          {
          if (jitConfig && jitConfig->getProcessorInfo())
             {
             buf = (TR_X86CPUIDBuffer *)jitConfig->getProcessorInfo();
             }
          else
             {
             buf = (TR_X86CPUIDBuffer *) malloc(sizeof(TR_X86CPUIDBuffer));
 
             if (!buf)
                memcpy(buf->_vendorId, "Poughkeepsie", 12); // 12 character dummy string (NIL term not used)
 
             buf->_processorSignature = 0;
             buf->_brandIdEtc = 0;
             buf->_featureFlags = 0x00000000;
             buf->_cacheDescription.l1instr = 0;
             buf->_cacheDescription.l1data  = 0;
             buf->_cacheDescription.l2      = 0;
             buf->_cacheDescription.l3      = 0;
             }
          }
       }
 
    return buf;
 #endif
    }
```
```
J9::X86::CPU::queryX86TargetCPUID() {
		static TR_X86CPUIDBuffer buf = { {'U','n','k','n','o','w','n','B','r','a','n','d'} };
		jitGetCPUID(&buf);
		return &buf;
}
```

**Comment by harryyu1994** [comment]
@mstoodle Hi Mark, this PR is ready for another review, thanks!

**Comment by 0xdaryl** [comment]
@mstoodle , I think this is ready to merge.

**Comment by charliegracie** [comment]
ping @mstoodle

**Comment by harryyu1994** [comment]
Allow me to rebase and run tests against it before merging.

**Comment by harryyu1994** [comment]
I think this PR will break a downstream project (OpenJ9), I will work on getting that PR approved today/tonight.

**Comment by mstoodle** [comment]
For the record, I believe that's this one: https://github.com/eclipse/openj9/pull/1613 right @harryyu1994 ?

**Comment by harryyu1994** [comment]
Will add old APIs back in for this change to prevent breaking downstream project.

**Comment by Leonardo2718** [comment]
@genie-omr build all

@mstoodle ping

**Comment by mstoodle** [comment]
For completeness: genie-omr build all

The openj9 PR that I suspect is co-dependent on this one cannot be merged right now due to (I believe) unrelated problems in the PR builds. Hopefully those issues will get cleared up soon...

**Comment by 0xdaryl** [comment]
@mstoodle, I believe both the downstream PR and this PR are ready to merge now.

**Comment by mstoodle** [comment]
OpenJ9 has asked that we hold off on this co-dependent merge until their release is closer to closed away (even though they've already split the release branch). Anyway, that request translated to "please hold off until Monday". Since this PR isn't urgent, I'm accepting the request and will revisit this PR on Monday.

**Comment by mstoodle** [comment]
Merging now along with dependent OpenJ9 PR
